### PR TITLE
refactor!: Remove `Mode` from `FrontendOptions`

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Resolve https://github.com/GreptimeTeam/greptimedb/issues/4399.

Remove `Mode` from `FrontendOptions` and determine whether it is Standalone or Distributed mode based on whether the Meta Client is configured.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Comment by sunng87** [comment]
Just realized that if user doesn't edit their config toml to remove this `mode`, will we throw an error for this "unknown" config item?

If so, this can be a breaking change. Perhaps we can keep it for a minor release and just change the behaviour

**Comment by zyy17** [comment]
> Just realized that if user doesn't edit their config toml to remove this `mode`, will we throw an error for this "unknown" config item?
> 
> If so, this can be a breaking change. Perhaps we can keep it for a minor release and just change the behaviour

Make sense. Let's keep it.

**Comment by tisonkun** [comment]
serde typically ignores unknown fields. Do we assert unknown fields to failure anywhere?

**Comment by zyy17** [comment]
> serde typically ignores unknown fields. Do we assert unknown fields to failure anywhere?

You are right. @sunng87, the `Mode` on the front end is not always necessary for our released version. When the user uses the current version, they don't need to modify their original config file and don't throw unrecoverable errors.

File: src/frontend/src/server.rs [hunk 1]
```diff
@@ -143,7 +143,17 @@ where
...
-        let runtime = match opts.mode {
+
+        // Determine whether it is Standalone or Distributed mode based on whether the meta client is configured.
+        let mode = {
+            if opts.meta_client.is_none() {
+                Mode::Standalone
+            } else {
+                Mode::Distributed
+            }
+        };
```
Comment (evenyag) [inline src/frontend/src/server.rs:154]: non-blocking: We don't need the outer brackets.
```suggestion
        let mode = if opts.meta_client.is_none() {
            Mode::Standalone
        } else {
            Mode::Distributed
        };
```

**Review by evenyag — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**